From: Alexander Sack <asac@debian.org>
Date: Wed, 27 Jan 2010 19:06:37 +0100
Subject: bz466250_att349521_fix_ftbfs_with_cairo_fb

#
# Ubuntu: none
# Upstream: https://bugzilla.mozilla.org/show_bug.cgi?id=466250
# Description: fix FTBFS with cairo 1.8.4 that is built with directfb,
#  triggering some new code, even when directfb was not wanted/detected
#  in configure
#
---
 mozilla/gfx/thebes/src/gfxASurface.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/mozilla/gfx/thebes/src/gfxASurface.cpp b/mozilla/gfx/thebes/src/gfxASurface.cpp
index 9eeb0ff..f278b15 100644
--- a/mozilla/gfx/thebes/src/gfxASurface.cpp
+++ b/mozilla/gfx/thebes/src/gfxASurface.cpp
@@ -56,8 +56,10 @@
 #endif
 
 #ifdef MOZ_DFB
+#ifdef MOZ_DFB
 #include "gfxDirectFBSurface.h"
 #endif
+#endif
 
 #ifdef CAIRO_HAS_QPAINTER_SURFACE
 #include "gfxQPainterSurface.h"
@@ -172,10 +174,12 @@ gfxASurface::Wrap (cairo_surface_t *csurf)
     }
 #endif
 #ifdef MOZ_DFB
+#ifdef MOZ_DFB
     else if (stype == CAIRO_SURFACE_TYPE_DIRECTFB) {
         result = new gfxDirectFBSurface(csurf);
     }
 #endif
+#endif
 #ifdef CAIRO_HAS_QPAINTER_SURFACE
     else if (stype == CAIRO_SURFACE_TYPE_QPAINTER) {
         result = new gfxQPainterSurface(csurf);
-- 
